packages/smooth_app/lib/helpers/analytics_helper.dart

```diff
@@ -0,0 +1,265 @@
+import 'dart:async';
+import 'dart:math';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+import 'package:matomo_forever/matomo_forever.dart';
+import 'package:openfoodfacts/model/Product.dart';
+import 'package:openfoodfacts/utils/OpenFoodAPIConfiguration.dart';
+import 'package:package_info_plus/package_info_plus.dart';
+import 'package:sentry_flutter/sentry_flutter.dart';
+import 'package:smooth_app/database/dao_int.dart';
+import 'package:smooth_app/database/local_database.dart';
+
+// TODO(m123): Check for user consent
+// TODO(m123): handle debug mode
+
+/// Helper for logging usage of core features and exceptions
+/// Logging:
+/// - Errors and Problems (sentry)
+/// - App start
+/// - Product scan
+/// - Product page open
+/// - Knowledge panel open
+/// - personalized ranking (without sharing the preferences)
+/// - search
+/// - external links
+
+const String _initAction = 'started app';
+const String _scanAction = 'scanned product';
+const String _productPageAction = 'opened product page';
+const String _knowledgePanelAction = 'opened knowledge panel page';
+const String _personalizedRankingAction = 'personalized ranking';
+const String _searchAction = 'search';
+const String _linkAction = 'opened link';
+
+/// The event category. Must not be empty. (eg. Videos, Music, Games...)
+const String eventCategory = 'e_c';
+
+/// Must not be empty. (eg. Play, Pause, Duration, Add
+/// Playlist, Downloaded, Clicked...)
+const String eventAction = 'e_a';
+
+/// The event name. (eg. a Movie name, or Song name, or File name...)
+const String eventName = 'e_n';
+
+/// Must be a float or integer value (numeric), not a string.
+const String eventValue = 'e_v';
+
+String latestSearch = '';
+
+Future<void> initSentry({Function()? appRunner}) async {
```

Comment:
I know what I wrote yesterday about the useless constructor; let me refine my comment - we don't need a _public_ empty constructor.
Ok, lint checks complain about static only class, so maybe a static class with a private constructor would do the trick.
Because those methods in the wild, especially the field `String latestSearch` with such a common name, that doesn't look good.
OR, what I would suggest is to keep that file as is, but to systematically `import ... as` this file (I don't know the correct term): `import '.../analytics_helper.dart' as analytics;`. Easier to read for reviewer than to try to find out where the hell this `latestSearch` comes from.

Reply:
I am also not a fan of global functions or especially values that you can not identify, but I am actually not a fan of `import ... as ...` so I put everything back into a static only class with a private constructor.

Reply:
My OOP tendencies thank you for that!